headlamp-k8s/headlamp
# WIP: Aggregated view

This is a preview of the aggregated views I have been working on.
TBH, I don't like how the code is turning, especially the apiProxy related code. This code was already very complicated to maintain ever since I added the API multi version support and these changes just make it even more complicated.

Probably a full refactor of how we run the API calls from the frontend is a good idea but I still wanted to share these changes and have the team take a look.

Beware that this is very WIP, including:
* Temporary method names
* Not aggregated metrics/charts yet
* Going into a resource details and back to the multi cluster view sometimes goes back only to a single cluster (the one where the resource lives)
* A cluster column is present in all list views but doesn't disappear when we're only in a single cluster
* Adding a resource while in a multi-cluster context not implemented yet

How to test:
1. Go into a cluster view and edit the URL to be the name of multiple clusters separated by +, example `minikube+mycluster1+mycluster2`
2. Notice that the version number (bottom left) now displays either 1 version (if all clusters have the same version), 2 versions (if only 2 different versions exist among the clusters), or `version+...` if more than 2 different versions exist among the clusters.
